src/v/storage/segment_utils.cc

```diff
@@ -578,6 +580,62 @@ ss::future<ss::lw_shared_ptr<segment>> make_concatenated_segment(
       std::nullopt);
 }
 
+ss::future<> write_concatenated_compacted_index(
+  std::filesystem::path target_path,
+  std::vector<ss::lw_shared_ptr<segment>> segments,
+  compaction_config cfg) {
+    if (segments.empty()) {
+        co_return;
+    }
+    std::vector<compacted_index_reader> readers;
+    readers.reserve(segments.size());
+
+    for (auto& s : segments) {
+        const auto path = compacted_index_path(s->reader().filename().c_str());
```

Comment:
will these conflict with existing file names?

Reply:
no, it is entirely new file name, the same as for segment.